app/lib/activitypub/tag_manager.rb

```diff
@@ -203,6 +203,39 @@ def uri_to_local_id(uri, param = :id)
     path_params[param]
   end
 
+  def uri_to_local_account_params(uri)
+    return unless local_uri?(uri)
+
+    path_params = Rails.application.routes.recognize_path(uri)
+
+    # TODO: handle numeric IDs
+    case path_params[:controller]
+    when 'accounts'
+      [:username, path_params[:username]]
+    when 'instance_actors'
+      [:id, -99]
+    end
+  end
+
+  def uri_to_local_account(uri)
+    param, value = uri_to_local_account_params(uri)
+
+    case param
+    when :username
+      Account.find_local(value)
+    when :id
+      Account.find_by(id: value)
+    end
+  end
```

Comment:
Could this not just call `#uris_to_local_accounts`? (Something like `uris_to_local_accounts([uri]).first` might work.)

I am unsure about having two methods that mostly do the same thing but in subtly different ways.

Reply:
I wanted to avoid the overhead of array creation and traversal, but it's probably premature optimization.

Reply:
This also begs the question if this is something YJIT can optimize away (I really have no idea).

Reply:
To be clear: I was not suggesting removing the method altogether, just making it a one-liner. But since it was only used in that one place, I can see how removing it makes sense as well (or maybe even more).